mbq/atomiq/management/commands/atomic_run_consumer.py

```diff
@@ -29,23 +31,66 @@ def __init__(self, *args, **kwargs):
         signal.signal(signal.SIGTERM, self.signal_handler.handle_signal)
         signal.signal(signal.SIGQUIT, self.signal_handler.handle_signal)
 
+        self.queues = {
+            constants.QueueType.SNS: {
+                'model': models.SNSTask,
+                'consumer_class': consumers.SNSConsumer,
+            },
+            constants.QueueType.SQS: {
+                'model': models.SQSTask,
+                'consumer_class': consumers.SQSConsumer,
+            },
+            constants.QueueType.CELERY: {
+                'model': models.CeleryTask,
+                'consumer_class': consumers.CeleryConsumer,
+            },
+        }
+
     def add_arguments(self, parser):
-        parser.add_argument('--queue', required=True)
+        queue_type_choices = [c[0] for c in constants.QueueType.CHOICES]
+        parser.add_argument('--queue', required=True, choices=queue_type_choices)
         parser.add_argument('--celery-app', required=False)
 
+    @utils.debounce(minutes=15)
+    def cleanup_old_tasks():
+        days_to_keep_old_tasks = constants.DEFAULT_DAYS_TO_KEEP_OLD_TASKS
+        time_to_delete_before = arrow.utcnow().shift(days=-days_to_keep_old_tasks)
+
+        model = self.queues[options['queue']]['model']
+        model.objects.filter(
+            state__in=[constants.TaskStates.SUCCEEDED, constants.TaskStates.DELETED],
+            created_at__lt=time_to_delete_before.datetime,
+        ).delete()
+
+    @utils.debounce(seconds=15)
+    def collect_metrics(self):
+        queue_type = options['queue']
+        model = self.queues[queue_type]['model']
+
+        state_counts = collections.Counter(model.objects.values_list('state', flat=True))
+        for state, count in state_counts.items():
+            _collector.gauge(
+                'state_total',
+                count,
+                tags={'state': state, 'queue_type': queue_type},
+            )
+
     def handle(self, *args, **options):
         try:
-            queue_name = options['queue']
-            if queue_name == 'sns':
-                consumer = consumers.SNSConsumer()
-            if queue_name == 'sqs':
-                consumer = consumers.SQSConsumer()
-            if queue_name == 'celery':
-                consumer = consumers.CeleryConsumer(celery_app=options['celery_app'])
+            queue_type = options['queue']
```

Comment:
Nit: We can use this decorator instead of indenting the whole handle function in a try/except:
(just added this in invoicing yesterday)
```
def send_errors_to_rollbar(func):

    def wrapper(*args, **kwargs):
        try:
            func(*args, **kwargs)
        except Exception:
            rollbar.report_exc_info()
            raise

    return wrapper
```

```
@send_errors_to_rollbar
def handle(...):
```

Reply:
i'm gonna be looking a bit more into that in the next PR.